### PR TITLE
Feat/token inject: Injecting token

### PR DESCRIPTION
**Pull request by GraceRuan**

### CODE REVIEW

**Comment by mbystedt** [comment]
Missing helm/broker-app/values/test.yaml

File: helm/broker-app/templates/deploymentconfig.yaml [hunk 1]
```diff
@@ -27,6 +27,18 @@ spec:
       labels:
         app: {{ .Values.global.name }}
         deploymentconfig: {{ .Values.global.name }}
+    annotations:
+        vault.hashicorp.com/agent-inject: 'true'
+        vault.hashicorp.com/agent-inject-secret-broker-token: a03c8f-{{ .Values.global.envlable }}/{{ .Values.global.envtag }}-token
+        vault.hashicorp.com/agent-inject-template-broker-token: |
+          {{`{{- with secret `}}"a03c8f-{{ .Values.global.envlable }}/{{ .Values.global.envtag }}-token" {{` }}
```
Comment (mbystedt) [inline helm/broker-app/templates/deploymentconfig.yaml:34]: License plate is a quasi-secret. Extract from: .Release.Namespace

https://helm.sh/docs/chart_template_guide/builtin_objects/

File: helm/broker-app/values.yaml [hunk 2]
```diff
@@ -4,6 +4,8 @@
 global:
   name: "nr-broker-app"
   environment: "development"
+  envtag: "dev"
+  envlable: "nonprod"
```
[the remark this answers is not in the document]
Comment (mbystedt) [reply helm/broker-app/values.yaml:8]: envtag / envlable could be extracted from global.environment. May need to add a helper function anyway to deal with removing the license plate.

See: https://helm.sh/docs/chart_template_guide/function_list/#regular-expressions

**Review by mbystedt — CHANGES_REQUESTED** [review]

File: helm/broker-app/templates/deploymentconfig.yaml [hunk 3]
```diff
@@ -27,6 +27,18 @@ spec:
...
+          export BROKER_TOKEN="{{ .Data.data.BROKER_TOKEN }}"
+          export VAULT_TOKEN="{{ .Data.data.VAULT_TOKEN }}"
+          {{ end `}} }}       
+        vault.hashicorp.com/agent-inject-token: 'true'
+        vault.hashicorp.com/auth-path: auth/k8s-CLUSTER
+        vault.hashicorp.com/namespace: {{ .Release.Namespace }}
+        vault.hashicorp.com/role: a03c8f-{{ .Values.global.envlable }}
```
Comment (andrwils) [inline helm/broker-app/templates/deploymentconfig.yaml]: Should we be extracting all license plate references from .Release.Namespace?

Comment (GraceRuan) [reply helm/broker-app/templates/deploymentconfig.yaml]: Yes, all license plate has been extracted from .Release.Namespace with the updated PR.

**Pull request closed without merging**